### PR TITLE
Polish response method deserialization

### PR DESCRIPTION
**Pull request by VeryEarly**

### CODE REVIEW

File: powershell/cmdlets/class.ts [hunk 1]
```diff
@@ -1009,7 +1021,7 @@ export class CmdletClass extends Class {
                     // it's just a nested array
                     const p = getVirtualPropertyFromPropertyName(schema.language.csharp?.virtualProperties, valueProperty.serializedName);
                     if (p) {
-                      yield $this.WriteObjectWithViewControl(`(await response).${p.name}`, true);
+                      yield $this.WriteObjectWithViewControl(`${result.value}.${p.name}`, true);
```
Comment (BethanyZhou) [inline powershell/cmdlets/class.ts:1024]: No sure if result.value is valid. result may be a primary type.

Comment (VeryEarly) [reply powershell/cmdlets/class.ts:1015]: result is a statement and result.value is string 'result'

File: powershell/cmdlets/class.ts [hunk 2]
```diff
@@ -505,16 +505,15 @@ export class CmdletClass extends Class {
...
+          yield If(`null != ${valueName}`, function* () {
+            yield If(`0 == _responseSize && 1 == ${valueName}.Count`, function* () {
+              yield `_firstResponse = ${valueName}[0];`;
               yield '_responseSize = 1;';
             });
             yield Else(function* () {
               yield $this.FlushResponse(false);
               yield 'var values = new System.Collections.Generic.List<System.Management.Automation.PSObject>();';
-              yield ForEach('value', 'outputObjects', function* () {
+              yield ForEach('value', valueName, function* () {
```
[the remark this answers is not in the document]
Comment (VeryEarly) [reply powershell/cmdlets/class.ts:516]: 'valueName' itself is a string, no concatenatation needed

**Review by BethanyZhou — APPROVED** [review]

**Review by dolauli — APPROVED** [review]
LGTM

**Review by dolauli — APPROVED** [review]
LGTM

**Pull request merged**